src/main/ruby/core/rbconfig.rb

```diff
@@ -149,6 +149,7 @@ module RbConfig
     cc             = "#{clang} -I#{sulong_include}"
     cpp            = cc
     cflags         = ['-Werror=implicit-function-declaration',  # To make missing C ext functions very clear
+                      '-Wno-unknown-warning-option',            # If we're on an earlier version of clang without a warning option, ignore it
```

Comment:
Is it needed? This might hide flags we add with a typo for instance.
If it's only a few extra lines of stderr output in CI I think it's fine.
But probably not so nice for the end-user.